### PR TITLE
Costy btanch

### PR DESCRIPTION
**Pull request by costy619**

### CODE REVIEW

**Review by Elenaadochitei — APPROVED** [review]

File: src/main/java/com/example/demo/ControllerBariera.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public DetaliiBariera getOneUser(@PathVariable("id") String id) {
     }
 
     @PostMapping("/postNewUser/{id}")
-    public DetaliiBariera postNewUser(@RequestBody DetaliiBariera detaliiBarieraList,@PathVariable("id") String id) throws Exception {
+    public DetaliiBariera postNewUser(@PathVariable("id") String id,@RequestBody DetaliiBariera detaliiBarieraList) throws Exception {
```
Comment (ionutvacariu) [inline src/main/java/com/example/demo/ControllerBariera.java:40]: please format code here

File: src/main/java/com/example/demo/ControllerBariera.java [hunk 2]
```diff
@@ -89,9 +88,10 @@ public LoginInfo getConfirmationOfUser(@RequestParam(value = "username") String
         String decodedString = (d.DecodingField(username));
         String decodedString1 = (d.DecodingField(pass));
         boolean isActive =connectExample.getConfirmation(decodedString,decodedString1);
+        String id=connectExample.id;
         LoginInfo log = new LoginInfo();
         log.logOn(isActive);
```
Comment (ionutvacariu) [inline src/main/java/com/example/demo/ControllerBariera.java:93]: why use logon ? return LoginInfo from getConfirmation()

**Review by ionutvacariu — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/ConnectLogInData.java [hunk 3]
```diff
@@ -8,21 +8,21 @@
...
-   static boolean getConfirmationOfUser=false;
+   String id;
+   boolean getConfirmationOfUser=false;
     public boolean getConfirmation(String username,String pass) throws Exception {
         try (Connection connection = DriverManager.getConnection(url, user, password);) {
             System.out.println("Connecting to database...");
             Statement statement = connection.createStatement();
             String sql;
-
             sql = "SELECT id, username, pass FROM USERS where username='"+username+"' and pass='"+pass+"'";
```
Comment (Elenaadochitei) [inline src/main/java/com/example/demo/ConnectLogInData.java:18]: why do you use sql injection?

File: src/main/java/com/example/demo/ConnectLogInData.java [hunk 4]
```diff
@@ -8,29 +8,21 @@
...
+   boolean getConfirmationOfUser=false;
     public boolean getConfirmation(String username,String pass) throws Exception {
         try (Connection connection = DriverManager.getConnection(url, user, password);) {
             System.out.println("Connecting to database...");
-
             Statement statement = connection.createStatement();
-
             String sql;
-            sql = "SELECT username, pass FROM USERS ";
+            sql = "SELECT id, username, pass FROM USERS where username='"+username+"' and pass='"+pass+"'";
```
Comment (Elenaadochitei) [inline src/main/java/com/example/demo/ConnectLogInData.java:18]: The code is vulnerable to SQL injections.

**Pull request closed without merging**